ccmlib/common.py

```diff
@@ -918,22 +918,17 @@ def _update_java_version(current_java_version, current_java_home_version,
     build_versions = [8]
     run_versions = [8]
 
-    info(
-        '{}: current_java_version={}, current_java_home_version={}, jvm_version={}, for_build={}, cassandra_version={}, install_dir={}, env={}'
-        .format(info_message, current_java_version, current_java_home_version, jvm_version, for_build,
-                cassandra_version, install_dir, env))
-
     # this won't work with DSE versions
-    if cassandra_version >= '4.2':
+    if cassandra_version >= '5.0':
         if os.path.exists(os.path.join(install_dir, 'build.xml')):
             build_versions = get_supported_jdk_versions(install_dir)
             run_versions = get_supported_jdk_versions(install_dir)
         else:
             # binary installs we don't know which jdks are supported any more
-            build_versions = [current_java_version]
-            run_versions = [current_java_version]
+            build_versions = [11, 17]
+            run_versions = [11, 17]
```

Comment:
this introduces a maintenance point.  every major version that bumps the supported jdks we break here, and have to go fix it.

my understanding is the node will fail-fast (won't start) with the appropriate message, so i'd rather not introduce these two lines.

Reply:
Without this, automatic jvm setup works up to 4.1; with this, automatic jvm works up to 5.x. 

We already decided to hardcode the supported Java versions - see the code for 4.x and previous versions.

This simple change only makes is less problematic to work with Cassandra 5.x.

Reply:
> We already decided to hardcode the supported Java versions - see the code for 4.x and previous versions.
> This simple change only makes is less problematic to work with Cassandra 5.x.


I see.

Is there really no way, for binary versions, that we can detect the supported versions without hardcoding it … ?

```
curl -s https://raw.githubusercontent.com/apache/cassandra/cassandra-5.0-beta1/build.xml | grep 'property\s*name="java.supported"' |sed -ne 's/.*value="\([^"]*\)".*/\1/p'
```

Reply:
Binary versions do not have build.xml; I think we should maybe store the supported java versions in some file; but on the other hand - this is really not a big deal to update CCM every couple of years for the supported java versions

Reply:
it can be done without the network too, as `java_versions_supported` is also found in `bin/cassandra.in.sh`
https://github.com/apache/cassandra/blob/trunk/bin/cassandra.in.sh#L117

Reply:
this is fixed now (a few lines above).  bin/cassandra.in.sh is only found in trunk, so 5.0 still needs the hardcode.